apps/src/Sounds.js

```diff
@@ -328,11 +378,76 @@ Sounds.prototype.playURL = function(url, playbackOptions) {
   }
 };
 
+/**
+ * @param {Promise} promise the promise that will resolve to the ArrayBuffer bytes to be played
+ * @param {object} playbackOptions config for the playing of the sound.
+ * @param {object} cacheParams optional parameters for caching the result of promise
+ */
+Sounds.prototype.addPromiseToSpeechQueue = function(
+  promise,
+  playbackOptions,
+  cacheParams = null
+) {
+  this.speechQueue.push({
+    promise: promise,
+    playbackOptions: playbackOptions,
+    cacheParams: cacheParams
+  });
+  this.checkSpeechQueue();
+};
+
+Sounds.prototype.checkSpeechQueue = async function() {
+  if (!this.speechPlaying && this.speechQueue.length > 0) {
+    this.speechPlaying = true;
+    let nextSpeech = this.speechQueue.shift();
+    let bytes = await nextSpeech['promise'];
```

Comment:
nit: prefer `nextSpeech.promise` over `nextSpeech['promise']` here and elsewhere